tests/lms/services/group_info_upsert_test.py

```diff
@@ -0,0 +1,100 @@
+from unittest import mock
+
+import pytest
+
+from lms.models import ApplicationInstance, GroupInfo
+from lms.services.group_info_upsert import GroupInfoUpsert
+
+
+class TestGroupInfoUpsert:
+    def test_it_adds_a_new_GroupInfo_if_none_exists(
+        self, db_session, group_info_upsert
+    ):
+        group_info_upsert(
+            "TEST_AUTHORITY_PROVIDED_ID",
+            "TEST_CONSUMER_KEY",
+            context_title="TEST_TITLE",
+            context_label="TEST_LABEL",
```

Comment:
I was thinking that this might be a good place for sentinels, but sentinels are not comparable with their stringification, and probably not serialisable to the DB anyway.

I wonder if we could fixture out settings here so we have:

```
@pytest.fixture etc..
settings = {
    authority_provided_id="TEST_AUTHORITY_PROVIDED_ID",
   ...
}
...
group_info_upsert(**settings)
...
assert group_info.consumer_key == settings.consumer_key
```
Rather than having the string repetition